libpod/container_internal.go

```diff
@@ -964,21 +975,23 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 			return nil, errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to add additional groups", c.ID())
 		}
 		for _, group := range c.config.Groups {
-			_, gid, err := chrootuser.GetUser(c.state.Mountpoint, strconv.Itoa(int(g.Spec().Process.User.UID))+":"+group)
+			gid, err := chrootuser.GetGroup(c.state.Mountpoint, group)
 			if err != nil {
 				return nil, err
 			}
-			g.AddProcessAdditionalGid(uint32(gid))
+			g.AddProcessAdditionalGid(gid)
 		}
 	}
 
-	// Look up and add groups the user belongs to
-	groups, err := chrootuser.GetAdditionalGroupsForUser(c.state.Mountpoint, uint64(g.Spec().Process.User.UID))
-	if err != nil {
-		return nil, err
-	}
-	for _, gid := range groups {
-		g.AddProcessAdditionalGid(gid)
+	// Look up and add groups the user belongs to if the user was defined by a username
+	if !userIsInt {
```

Comment:
Nope, if the user is just an int, we shouldn't look it up in /etc/passwd - https://github.com/projectatomic/libpod/issues/631#issuecomment-382027834

Reply:
Alright, SGTM then

Reply:
Can we move this block inside the `if c.config.User != ""`, then? We can't have have a UID user is user is not set at all. It would also let us get rid of the `var` block.

Reply:
but can we lookup `/etc/passwd` by UID instead of the username?  `GetAdditionalGroupsForUser` already expects the UID so should be safe to pass the numeric value to `GetAdditionalGroupsForUser`

Reply:
Yes good point.  We should look up the user by UID, although not fail if the UID does not exists.

Reply:
Fixed